backend/controller/controller.go

```diff
@@ -1150,14 +1151,16 @@ func (s *Service) getDeployment(ctx context.Context, key string) (*model.Deploym
 }
 
 // Return or create the RunnerService and VerbService clients for a Runner.
-func (s *Service) clientsForRunner(key model.RunnerKey, endpoint string) clients {
+func (s *Service) clientsForRunner(ctx context.Context, key model.RunnerKey, endpoint string) clients {
 	clientItem := s.clients.Get(key.String())
-	if clientItem != nil {
+	if clientItem != nil && clientItem.Value().created_at.Add(time.Minute*5).After(time.Now()) {
 		return clientItem.Value()
 	}
+	log.FromContext(ctx).Tracef("Creating new grpc clients for %s at %s", key, endpoint)
 	client := clients{
-		runner: rpc.Dial(ftlv1connect.NewRunnerServiceClient, endpoint, log.Error),
-		verb:   rpc.Dial(ftlv1connect.NewVerbServiceClient, endpoint, log.Error),
+		runner:     rpc.Dial(ftlv1connect.NewRunnerServiceClient, endpoint, log.Error),
```

Comment:
Does this actually create a new client in terms of the underlying connections? rcp.Dial seems to use a static HTTP client, and AFAIK go clients manage their own connection pools, so I think we may have an implicit connection cache.

Reply:
https://github.com/TBD54566975/ftl/blob/787444ee9d8ec79cec02f645dfaea2c6726cdcda/internal/rpc/rpc.go#L32 is the client I am talking about.

Reply:
```
// The [Client.Transport] typically has internal state (cached TCP
// connections), so Clients should be reused instead of created as
// needed.
```
It seems so yes. But it also seems that having one http.Client is correct usage 🤔